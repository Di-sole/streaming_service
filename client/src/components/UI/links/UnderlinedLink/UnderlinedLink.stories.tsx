import { Meta, StoryObj } from '@storybook/react';
import Link from './UnderlinedLink';
import './UnderlinedLink.module.scss';

const meta: Meta<typeof Link> = {
    title: 'Links/UnderlinedLink',
    component: Link,
    decorators: [
        (Story) => (
            <div style={{ background: '#100e19' }}>
                <Story />
            </div>
        ),
    ],
};

export default meta;

type Story = StoryObj<typeof Link>;

export const UnderlinedLink: Story = {
    args: {
        text: 'Подчеркнутая ссылка',
    },
};
