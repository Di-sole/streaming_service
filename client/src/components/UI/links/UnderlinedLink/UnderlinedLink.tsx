import Link from 'next/link';
import styles from './UnderlinedLink.module.scss';

interface UnderlinedLinkProps {
    text: string;
    href?: string;
}

const UnderlinedLink = ({ text, href = '' }: UnderlinedLinkProps) => {
    return (
        <Link className={styles.link} href={href}>
            {text}
        </Link>
    );
};

export default UnderlinedLink;
