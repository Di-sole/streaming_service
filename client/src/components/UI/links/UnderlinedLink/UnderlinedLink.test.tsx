import { render, screen } from '@testing-library/react';
import UnderlinedLink from './UnderlinedLink';

describe('UNDERLINEDLINK SNAPSHOTS TESTS', () => {
    test('UnderlinedLink should not be changed', () => {
        render(<UnderlinedLink href="/" text="text" />);
        const link = screen.getByRole('link');
        expect(link).toMatchSnapshot();
    });
});
