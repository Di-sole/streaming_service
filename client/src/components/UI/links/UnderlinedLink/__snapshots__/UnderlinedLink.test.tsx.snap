// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UNDERLINEDLINK SNAPSHOTS TESTS UnderlinedLink should not be changed 1`] = `
<a
  class="link"
  href="/"
>
  text
</a>
`;
